pkg/kubectl/cmd/plugin.go

```diff
@@ -109,6 +112,15 @@ func NewCmdForPlugin(f cmdutil.Factory, plugin *plugins.Plugin, runner plugins.P
 			}
 
 			if err := runner.Run(plugin, runningContext); err != nil {
+				if exiterr, ok := err.(*exec.ExitError); ok {
+					// check for (and exit with) the correct exit code
+					// from a failed plugin command execution
+					if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
```

Comment:
Hrm, need to check if this works on Mac and Windows.

Reply:
Sure, will go ahead and test this on both

Reply:
Seems to work as expected on Windows:
![windows_exit_code](https://user-images.githubusercontent.com/3902875/31746299-b1c7a9c4-b434-11e7-981b-78233d6b3a08.png)

Will test on mac tomorrow and post screenshot for it.

Reply:
@fabianofranz works as expected on mac:
<img width="800" alt="screen shot 2017-10-20 at 5 05 13 pm" src="https://user-images.githubusercontent.com/3902875/31841937-c9663308-b5b9-11e7-8513-2bbd408f8fe0.png">

@sg00dwin thanks for helping me test on mac